src/tree_types.ml

```diff
@@ -0,0 +1,265 @@
+(** Miscellaneous types and utilities for decision trees. *)
+open Extra
+
+(** {3 Tree constructors} *)
```

Comment:
Define a module `Treecons` here, and rather use the name `t` for `treecons` so that it is accessed with `Treecons.t` everywhere. Then rename `tc_pp` into `pp` and `tc_compare` (which should just be a synonym of `Pervasives.compare`) into `compare`. Then you can use this module directly to instantiate `Map.Make` (see further comments).